.github/workflows/ci.yml

```
@@ -0,0 +1,226 @@
+        go-version: ['1.21', '1.22']
+        exclude:
+          # Only test latest Go on macOS to save CI minutes
+          - os: macos-latest
+            go-version: '1.21'
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
```

actionlint (1.7.7)

the runner of "actions/setup-go@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)